### PR TITLE
Add test to chocolate-doom

### PR DESCRIPTION
**Pull request by Git-Jiro**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Reference https://github.com/Homebrew/homebrew-core/issues/11898

### CODE REVIEW

**Comment by fxcoudert** [comment]
```
==> chocolate-doom; if [ $? -ne 255 ]; then exit -1; else exit 0; fi
Error: chocolate-doom: failed
An exception occurred within a child process:
  Timeout::Error: execution expired
```

**Comment by Moisan** [comment]
The test opens a window on which the button "ok" needs to be clicked to finish the test. A valid test should automatically finish without the user intervention.

**Comment by Git-Jiro** [comment]
After googling around it seems like this will never work. chocolate-doom is a GUI app and I have not found any way so far to test a GUI app running headless on macOS.

**Comment by Git-Jiro** [comment]
Is there any way to automate clicking the OK button using some automation tool which I could include into the test?

**Comment by Moisan** [comment]
Including an automation tool in the test is not the right way to go. We try to keep the dependencies as low as possible. `chocolate-doom` will always fail since it needs a Doom level that is under copyright. I think the right way to test it would be something like 
```assert_match /Chocolate Doom 3.0.0/, shell_output("#{bin}/chocolate-doom -nogui", 255)```

While this is not usually the preferred type of test, we can make an exception in this case.

File: Formula/chocolate-doom.rb [hunk 1]
```diff
@@ -46,4 +46,8 @@ def caveats; <<~EOS
       #{homepage}
   EOS
   end
+
+  test do
+    system "chocolate-doom; if [ $? -ne 255 ]; then exit -1; else exit 0; fi"
```
Comment (Moisan) [inline Formula/chocolate-doom.rb]:
```suggestion
    assert_match /Chocolate Doom 3.0.0/, shell_output("#{bin}/chocolate-doom -nogui", 255)
```

**Review by Moisan — COMMENTED** [review]
Same as last comment, with the new github suggestion feature :).

File: Formula/chocolate-doom.rb (hunk 1 above)
Comment (commitay) [reply Formula/chocolate-doom.rb]: `assert_match "Chocolate Doom #{version}"`

**Comment by Moisan** [comment]
Thank you @Git-Jiro !

**Pull request merged**